test/integration/sequelize.test.js

```diff
@@ -386,6 +386,40 @@ describe(Support.getTestDialectTeaser('Sequelize'), function() {
       }).to.throw(Error, 'Both `replacements` and `bind` cannot be set at the same time');
     });
 
+    if (Support.dialectIsMySQL()) {
+      it.only('properly adds and escapes replacement value', function () {
+        var logSql,
+            number  = 1,
+            date = new Date(),
+            string = 't\'e"st',
+            boolean = true,
+            buffer = new Buffer('t\'e"st');
+
+        date.setMilliseconds(0);
+        return this.sequelize.query({
+            query: 'select ? as number, ? as date,? as string,? as boolean,? as buffer',
+            values: [number, date, string, boolean, buffer]
+          }, {
+            type: this.sequelize.QueryTypes.SELECT,
+            logging: function(s) {
+              logSql = s;
+            }
+          }).then(function(result) {
+            var res = result[0] || {};
+            res.date = res.date && new Date(res.date);
+            res.boolean = res.boolean && true;
+            expect(res).to.deep.equal({
+              number : number,
+              date   : date,
+              string : string,
+              boolean: boolean,
+              buffer : buffer
+            });
+            expect(logSql.indexOf('?')).to.equal(-1);
+        });
+      });
+    }
```

Comment:
I wasn't sure if the format being returned will be identical on all dialects
and since I don' t have any other db install on my machine and  so couldn't run this test with other dialects and I didn't want to add test that might fail with other dialects.